experiments/yarn/src/main/scala/io/gearpump/experiments/yarn/client/LaunchCluster.scala

```diff
@@ -0,0 +1,205 @@
+  private val memory = akka.getString(APPMASTER_MEMORY).toInt
+  private val vcore = akka.getString(APPMASTER_VCORES).toInt
+
+
+  def submit(appName: String, packagePath: String): ApplicationId = {
+    LOG.info("Starting AM")
+
+    // first step, check the version, to make sure local version matches remote version
+    if (!packagePath.endsWith(".zip")) {
+      throw new IOException(s"YarnClient only support .zip distribution package, now it is ${packagePath}. Please download the zip " +
+        "package from website or use sbt assembly packArchiveZip to build one.")
+    }
+
+    if (!fs.exists(packagePath)) {
+      throw new IOException(s"Cannot find package ${packagePath} on HDFS ${fs.name}. ")
+    }
+
+    val rootEntry = rootEntryPath(zip = packagePath)
+
+    if (!rootEntry.contains(version)) {
+      throw new IOException(s"Check version failed! Local gearpump binary version $version doesn't match with remote path $packagePath")
+    }
+
+    val resource = Resource.newInstance(memory, vcore)
+    val appId = yarnClient.createApplication
+
+    // will upload the configs to HDFS home directory of current user.
+    LOG.info("Uploading configuration files to remote HDFS(under /user/<id>/.gearpumpapp_)...")
+    val configPath = uploadConfigToHDFS(appId)
+
+    val command = AppMasterCommand(akka, rootEntry, Array(s"-conf $configPath", s"-package $packagePath"))
+
+    yarnClient.submit(appName, appId, command.get, resource, queue, packagePath, configPath)
+
+    LOG.info("Waiting application to finish...")
+    val report = yarnClient.awaitApplication(appId, LaunchCluster.TIMEOUT_MILLISECONDS)
+    LOG.info(s"Application $appId finished with state ${report.getYarnApplicationState} at ${report.getFinishTime}, info: ${report.getDiagnostics}")
+    Console.println("================================================")
+    Console.println("==Application Id: " + appId)
+    appId
+  }
+
+  def saveConfig(appId: ApplicationId, output: String): Future[File] = {
+    val appMaster = appMasterResolver.resolve(appId)
+    val future = askActor[ActiveConfig](appMaster, GetActiveConfig(host)).map(_.config)
+    import scala.concurrent.duration._
+    future.map{ config =>
+      val out =  new File(output)
+      ClusterConfig.saveConfig(config, out)
+      out
+    }
+  }
+
+  private def uploadConfigToHDFS(appId: ApplicationId): String = {
+    // will use personal home directory so that it will not conflict with other users
+    val confDir = s"${fs.getHomeDirectory}/.gearpump_app${appId.getId}/conf/"
+
+    // copy config from local to remote.
+    val remoteConfFile = s"$confDir/gear.conf"
+    var out = fs.create(remoteConfFile)
+    var writer = new OutputStreamWriter(out)
+
+    val cleanedConfig = ClusterConfig.filterOutDefaultConfig(akka)
+
+    writer.write(cleanedConfig.root().render())
+    writer.close()
+
+    // save yarn-site.xml to remote
+    val yarn_site_xml = s"$confDir/yarn-site.xml"
```

Comment:
You will also upload yarn-site.xml under conf, not mentioned in the doc.
Also, command ```yarnClient``` contains ```/etc/hadoop/conf``` in it's classpath, if user's yarn conf is not in that folder, will it failed to locate the right Hdfs address?